presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/TestApplyDynamicFilters.java

```diff
@@ -0,0 +1,193 @@
+    }
+
+    @Test
+    public void testMultipleDynamicFilterConditions()
+            throws Exception
+    {
+        tester.assertThat(new ApplyDynamicFilters())
+                .on(p -> p.join(
+                        INNER,
+                        p.values(p.symbol("L_COL1", BIGINT), p.symbol("L_COL2", BIGINT)),
+                        p.values(p.symbol("R_COL1", BIGINT), p.symbol("R_COL2", BIGINT)),
+                        ImmutableList.of(
+                                new JoinNode.EquiJoinClause(new Symbol("L_COL1"), new Symbol("R_COL1")),
+                                new JoinNode.EquiJoinClause(new Symbol("L_COL2"), new Symbol("R_COL2"))),
+                        ImmutableList.of(new Symbol("L_COL1"), new Symbol("R_COL1")),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Assignments.of()))
+                .matches(
+                        join(
+                                INNER,
+                                ImmutableList.of(
+                                        equiJoinClause("L_COL1", "R_COL1"),
+                                        equiJoinClause("L_COL2", "R_COL2")),
+                                ImmutableMap.of("L_COL1", "R_COL1", "L_COL2", "R_COL2"),
+                                values("L_COL1", "L_COL2"),
+                                values("R_COL1", "R_COL2")));
+    }
+
+    @Test
+    public void testMultipleEquiConditionsSingleDynamicFilterCondition()
+            throws Exception
+    {
+        tester.assertThat(new ApplyDynamicFilters())
+                .on(p -> p.join(
+                        INNER,
+                        p.values(p.symbol("L_COL1", BIGINT), p.symbol("L_COL2", BIGINT)),
+                        p.values(p.symbol("R_COL1", BIGINT), p.symbol("R_COL2", BIGINT)),
+                        ImmutableList.of(
+                                new JoinNode.EquiJoinClause(new Symbol("L_COL1"), new Symbol("R_COL1")),
+                                new JoinNode.EquiJoinClause(new Symbol("L_COL2"), new Symbol("R_COL2"))),
+                        ImmutableList.of(new Symbol("L_COL1"), new Symbol("R_COL1")),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        Assignments.of(new Symbol("R_COL2_ALIAS"), new Symbol("R_COL2").toSymbolReference())))
+                .matches(
+                        join(
+                                INNER,
+                                ImmutableList.of(
+                                        equiJoinClause("L_COL1", "R_COL1"),
+                                        equiJoinClause("L_COL2", "R_COL2")),
+                                ImmutableMap.of("L_COL1", "R_COL1"),
+                                values("L_COL1", "L_COL2"),
+                                values("R_COL1", "R_COL2")));
+    }
+}
```

Comment:
This is unit tests that verifies only the apply part. There is a separate tests that verifies the remove part.